app/scripts/directives/blocks.js

```diff
@@ -169,6 +169,28 @@ angular
           ['selectedAnswer', 'otherAnswer'],
           updateAnswerValueDebounced,
         );
+
+        // Watch for changes to answer.value and update selectedAnswer accordingly
+        $scope.$watch('answer.value', function (newValue, oldValue) {
```

Comment:
This logic looks similar to this section. Could they be combined?

```js
const answerValue = $scope.answer ? $scope.answer.value : '';
if (
  $scope.block.content.otherOption &&
  $scope.block.content.otherOption.enabled &&
  answerValue &&
  !_.includes(_.map($scope.block.content.choices, 'value'), answerValue)
) {
  // An answer is present that isn't in the list of available choices, so it is an "Other" answer
  $scope.selectedAnswer = $scope.otherSentinel;
  $scope.otherAnswer = answerValue;
} else {
  $scope.selectedAnswer = answerValue;
  $scope.otherAnswer = '';
}
```

Reply:
Great simplification! I'll make the change